lib/grizzly/zwave/commands/configuration_set.ex

```diff
@@ -80,8 +80,14 @@ defmodule Grizzly.ZWave.Commands.ConfigurationSet do
     param_num = Command.param!(command, :param_number)
     size = Command.param!(command, :size)
     value = Command.param!(command, :value)
+    :ok = validate!(value, size)
     value_bin = <<value::signed-integer-size(size)-unit(8)>>
 
     <<param_num, size>> <> value_bin
   end
+
+  defp validate!(value, 1) when value in -128..127, do: :ok
+  defp validate!(value, 2) when value in -32768..32767, do: :ok
+  defp validate!(value, 4) when value in -2_147_483_648..2_147_483_647, do: :ok
+  defp validate!(_value, _size), do: :error
```

Comment:
```suggestion
```
Bang functions normally raise on bad parameters, so there shouldn't need to be an `:ok = ` check up above. One way to fix this is to delete the catch-all. I don't know how user visible that is, though. If it is user visible, then I'd try `ArgumentError` or whatever you all think is friendliest.

Could you add a unit test to verify that exceptions are raised?